src/store/selectors/stories/types.ts

```diff
@@ -0,0 +1,258 @@
+}
+
+/**
+ * Reference to find and render an image.
+ */
+export interface ImageReference {
+  imageUri: string;
+}
+
+/**
+ * Reference to visual media to dipslay.
+ */
+export type MediaReference = VideoReference | ImageReference;
+
+/**
+ * Types of bodies that the app has built-in logic to display.
+ */
+export enum BodyType {
+  MEDIA,
+  TEXT,
+  MINT_BASIC_INCOME,
+  TRUST_MEMBER
+}
+
+/**
+ * Reference to a body that the application independently decides how to
+ * display, like by displaying an icon.
+ */
+export interface PredeterminedBody {
+  predeterminedBodyType: BodyType;
+}
+
+export interface MediaBody {
+  type: BodyType.MEDIA;
+  media: MediaReference[];
+}
+
+export interface TextBody {
+  type: BodyType.TEXT;
+  text: string;
+}
+
+/**
+ * Body to display for a given story.
+ */
+export type StoryBody =
+  | TextBody
+  | MediaBody
+  | { type: BodyType.MINT_BASIC_INCOME }
+  | { type: BodyType.TRUST_MEMBER };
+
+export interface NextInChain {
+  direction: ChainDirection;
+  nextStoryContent: StoryContent;
+}
+
+/**
+ * The content of an Story. Missing some metadata that makes a complete,
+ * renderable Story.
+ */
+export interface StoryContent {
+  /**
+   * What members took action. Stored as an OrderedMap in order to not keep
+   * duplicates. Would use an OrderedSet, but Immutable.JS doesn't support
+   * custom equality functions, so Map simulates equality by id.
+   *
+   * TODO: potentially support other special members than basic income, treat it
+   * as an actual member/variation of it rather than a singleton
+   */
+  actors: OrderedMap<MemberId, Member> | typeof RAHA_BASIC_INCOME_MEMBER;
+  /**
+   * Description of the action they took. Currently rendered after the actor's
+   * display name in the format of a complete sentence.
+   */
+  description?: (string | CurrencyValue)[];
+  /**
+   * A larger, detailed body describing (in text) or representing (visually) the
+   * action.
+   *
+   * Currently, a body is required if there is to be further chained content. If
+   * that is too restrictive, we can relax that restriction, but it will likely
+   * require a visual redesign as well.
+   */
+  body?: {
+    bodyContent: StoryBody;
+    /**
+     * The next piece of content in the chain of activities.
+     */
+    nextInChain?: NextInChain;
+  };
+}
+
+/**
+ * A renderable link that directs a user to an action they can take.
+ */
+export interface ActionLink {
+  text: string;
+  destination: LinkDestination;
+}
+
+export enum CallToActionDataType {
+  TEXT = "TEXT",
+  LINK = "LINK"
+}
+
+export type CallToActionPiece =
+  | {
+      type: CallToActionDataType.TEXT;
+      data: (string | CurrencyValue)[];
+    }
+  | {
+      type: CallToActionDataType.LINK;
+      data: ActionLink;
+    };
+
+/**
+ * An invitation for a member to take action.
+ */
+export type CallToAction = CallToActionPiece[];
+
+/**
+ * Shape of the possible data that can comprise a story
+ */
+export interface StoryDataDefinition<
```

Comment:
hmm why are `StoryContent` and `StoryData` separated? If `StoryData` is  always Activity/Activities, can we call it `StoryActivity` instead?

Reply:
the `type` sets it apart from just a list of activities, since some activities can be intepreted in different ways—a mint operation is treated differently if it's a referral or a mint, a verify operation is treated differently if it's alone vs tied to a new member scenario. I wanted `StoryData` to unambiguously define how the actual visual content—the `StoryContent`—gets generated.

Reply:
added more docs to explain this.

Reply:
Thanks for the updated docs!